Overlapping pattern/lookup combinations

I quite recently ran in to a typo which `respx` doesn't pick up on. It's probably not really `respx`s responsibility but thought perhaps it could deserve a warning? Or even better if it'd raise an error.

```python
import httpx
import respx
mocker = respx.mock(base_url="http://some.where/", assert_all_called=False)
mocker.post(method__in=("GET", "PUT"), path="/here/")
with mocker:
    httpx.post("http://some.where/here/", json={"la": "la"})
```

__Raises:__

```
respx.models.AllMockedAssertionError: RESPX: <Request('POST', 'http://some.where/here/')> not mocked!
```
Choosing any of the methods `POST`, `GET` or `PUT`. The same error is raised since there's really nothing that could possibly match more than 1 method.

This is the constraint on the route:
```
<Route <Scheme eq 'http'> AND <Host eq 'some.where'> AND <Method eq 'POST'> AND <Method in ('GET', 'PUT')> AND <Path eq '/here/'>>
```

--- a/respx/router.py
+++ b/respx/router.py
@@ -159,7 +159,7 @@ class Router:
             pattern_keys = {p.split("__", 1)[0] for p in lookups.keys()}
             if "method" in pattern_keys:
                 raise TypeError("Got multiple values for pattern 'method'")
-            if url and "url" in pattern_keys:
+            elif url and "url" in pattern_keys:
                 raise TypeError("Got multiple values for pattern 'url'")
 
         return self.route(method=method, url=url, name=name, **lookups)

--- a/respx/router.py
+++ b/respx/router.py
@@ -156,7 +156,7 @@ class Router:
     ) -> Route:
         if lookups:
             # Validate that lookups doesn't contain method or url
-            pattern_keys = {p.split("__")[0] for p in lookups.keys()}
+            pattern_keys = {p.split("__", 1)[0] for p in lookups.keys()}
             if "method" in pattern_keys:
                 raise TypeError("Got multiple values for pattern 'method'")
             if url and "url" in pattern_keys:

--- a/respx/router.py
+++ b/respx/router.py
@@ -154,6 +154,14 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
+        if lookups:
+            # Validate that lookups doesn't contain method or url
+            pattern_keys = {p.split("__")[0] for p in lookups.keys()}
+            if "method" in pattern_keys:
+                raise TypeError("Got multiple values for pattern 'method'")
+            if url and "url" in pattern_keys:
+                raise TypeError("Got multiple values for pattern 'url'")
+
         return self.route(method=method, url=url, name=name, **lookups)
 
     def get(
@@ -163,7 +171,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="GET", url=url, name=name, **lookups)
+        return self.request(method="GET", url=url, name=name, **lookups)
 
     def post(
         self,
@@ -172,7 +180,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="POST", url=url, name=name, **lookups)
+        return self.request(method="POST", url=url, name=name, **lookups)
 
     def put(
         self,
@@ -181,7 +189,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="PUT", url=url, name=name, **lookups)
+        return self.request(method="PUT", url=url, name=name, **lookups)
 
     def patch(
         self,
@@ -190,7 +198,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="PATCH", url=url, name=name, **lookups)
+        return self.request(method="PATCH", url=url, name=name, **lookups)
 
     def delete(
         self,
@@ -199,7 +207,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="DELETE", url=url, name=name, **lookups)
+        return self.request(method="DELETE", url=url, name=name, **lookups)
 
     def head(
         self,
@@ -208,7 +216,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="HEAD", url=url, name=name, **lookups)
+        return self.request(method="HEAD", url=url, name=name, **lookups)
 
     def options(
         self,
@@ -217,7 +225,7 @@ class Router:
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
-        return self.route(method="OPTIONS", url=url, name=name, **lookups)
+        return self.request(method="OPTIONS", url=url, name=name, **lookups)
 
     def record(
         self,

--- a/tests/test_router.py
+++ b/tests/test_router.py
@@ -394,6 +394,14 @@ def test_rollback():
     assert route.return_value is None
 
 
+def test_multiple_pattern_values_type_error():
+    router = Router()
+    with pytest.raises(TypeError, match="Got multiple values for pattern 'method'"):
+        router.post(method__in=("PUT", "PATCH"))
+    with pytest.raises(TypeError, match="Got multiple values for pattern 'url'"):
+        router.get("https://foo.bar", url__regex=r"https://example.org$")
+
+
 def test_routelist__add():
     routes = RouteList()
 

